src/main/java/bogwarden/patches/TrapPatches.java

```diff
@@ -32,7 +32,17 @@ public static void Insert(AbstractPlayer __instance, DamageInfo info) {
         }
     }
 
-    
+    // safety net for cross-mod compatibility
+    @SpirePatch(clz=AbstractPlayer.class, method="updateCardsOnDamage")
```

Comment:
looks like this patch will always occur before the usual TriggerOnDamage one, making the trap target always random.
i'd suggest removing the original triggerondamage insert patch, and in its existing prefix patch storing the damageinfo for use later in this patch for its target.